### ISSUE TITLE
Would it be difficult to port the keybindings over to VIM?

### ISSUE THREAD

**Issue by gytis-ivaskevicius** [mawww/kakoune#4106]
I was thinking if there is some somewhat automagic way of porting the keybindings over to vim? Maybe one could autogenerate em from something? or at least do it partially?

Reasoning:
The massive advantage of VIM is that extension for it exists for pretty much any editor you can think of, some of which support straight up vimscript and/or neovim as a backend. Kakoune keybindings port as such would allow using kakoune like keybindings on other editors and would make it a viable option for me and other like-minded individuals to learn and migrate to

**Comment by eraserhd** [comment]
I think the big problem would be multiple selections.  It should be somewhat easy to emulate the rest.

**Comment by gytis-ivaskevicius** [comment]
As for the initial question - can we autogenerate at least some part of it? Like how does one list existing kakoune shortcuts? Maybe something else can be done to simplify task as such?
in other words - is there anything that would help a user that has next to no kak experience to do so?

**Comment by Screwtapello** [comment]
I don't think it would be practical to properly emulate Kakoune in Vim. I don't think Vim lets you select text by coordinates (like Kakoune's `:select`), and I don't think the "anchor" position of visual mode is easily available. I'm not even sure the cursor position is easily available, although I'm sure it must be somehow. I suspect you'd have to re-implement all the text-selection logic in VimScript, including jumping to the correct line, entering Visual mode at the calculated anchor position, and moving to the cursor to the calculated cursor position after every movement command. Even if it worked, it probably wouldn't be pleasant to use.

And if it's not practical to implement on top of Vim, it almost certainly wouldn't be practical to implement on top of a Vim plugin for some other editor.

**Comment by alexherbo2** [comment]
You can call Kakoune from Vim to leverage the use of multiple selections.

See [kakoune.vim].

[kakoune.vim]: https://github.com/alexherbo2/kakoune.vim

**Comment by gytis-ivaskevicius** [comment]
> And if it's not practical to implement on top of Vim, it almost certainly wouldn't be practical to implement on top of a Vim plugin for some other editor.

Thats unfortunate, by any chance maybe anyone has any other bright ideas so that one would be able kakoune more or less everywhere? 
I assume kakoune server/client protocol is nowhere near similar to neovim - if that's the case, it rules out forking extensions and tweaking it to support kak instead
